### ISSUE TITLE
Use wxWidgets for the GUI

### ISSUE THREAD

**Issue by clayton256** [lanurmi/efte#102]
I would like to suggest doing a wxWidgets port of efte/xfte. This way much of the effort put into developing for individual platforms could be put into a single code base. #

**Comment by lanurmi** [comment]
Using wxWidgets would certainly help creating GUI versions for platforms that don't have one yet. However, I would keep wxWidgets as an optional dependency; making it a requirement for compiling efte would be pretty annoying in some scenarios.

**Comment by DKnoto** [comment]
I'm currently using CodeBlocks on Linux. It is based on wxWidgets. Compared with Qt Creator it is horribly slow and ugly :(
I would prefer to eFTE went in the direction such as the sublime text.

**Comment by lanurmi** [comment]
The toolkit alone doesn't cause slowness or ugliness; it is how the toolkit is being used.

**Comment by DKnoto** [comment]
Not exactly. wxWidget works on X11, Gtk or Qt. On X11 is very ugly. On Gtk
is very buggy and slowly.
On Qt I don't know. I did not use it.

2017-02-19 22:24 GMT+01:00 Lauri Nurmi <notifications@github.com>:

**Comment by lanurmi** [comment]
> Not exactly. wxWidget works on X11, Gtk or Qt. On X11 is very ugly. On Gtk is very buggy and slowly. On Qt I don't know. I did not use it.

I have been using wxWidgets at work for more than 10 years, so I know a thing or two about it. And it is not "very buggy and slowly" on Gtk. Sure it is possible to make buggy and slow programs with it, but again, it is about how it is being used.

**Comment by clayton256** [comment]
I side with lanurmi on this. I've been using wxWidgets off and on for more than 15 years on multiple platforms and been very happy with it's native look and feel and it's great performance (including some animation). I also use a number of application based on wxWidgets and have not experienced any slowness, ugliness or bugs.
wxWidgets uses the native GUI (admittedly there are choices to make on *nix so it introduces a little complication that Mac, Win, etc doesn't have, but what commonly used *nix doesn't have GTK these days?). From the wxWigets site, "...and unlike other cross-platform toolkits, wxWidgets gives applications a truly native look and feel because it uses the platform's native API rather than emulating the GUI." 
Besides, Qt, I feel, is much harder to code for. I abandoned KDE because I didn't like Qt. I will not install an application based on Qt. But that's just MHO and not meant start a flame war.

**Comment by DKnoto** [comment]
Generaly you're right. I don't working in this toolkit. I am only using
CodeBlocks for few years
and testing wxWidgets from time to time.

I like Qt and I've been using in since 1995-6 but this toolkit isn't ideal
also.

2017-02-20 15:50 GMT+01:00 Mark Clayton <notifications@github.com>:
